### DISCUSSION TITLE
Uso de `.gitkeep`

### DISCUSSION THREAD

**Discussion by JavierLianoRioz** [mmasias/24-25-PRG1#701] [Show and tell]
# ARCHIVO `.gitkeep`
He descubierto una cosa muy útil a la hora de subir carpetas vacías a un repositorio.

## Problema
Cuando queremos subir una carpeta vacía a un repositorio nos encontramos con el problema de que git no gestiona/almacena carpetas sin contenido dentro. 

> [!NOTE]
> Por esta razón hemos usado archivos temporales para este mismo propósito. También hemos hecho uso de archivos `README.md` y parecidos que nos dan la soltura de dar unas instrucciones básicas de para lo que sirve la carpeta. Pero aún así hay veces que no se necesitan instrucciones y solamente la quieres subir vacía.

## Solución
Para solucionar esto, existe el archivo `.gitkeep` que es una nomenclatura generalizada en el mundo de git. Nos permite subir una carpeta a un repositorio y que esta este vacía. Es lo mismo que antes, pero es la alternativa nativa de git para sobrepasar las limitaciones de git.

> [!TIP]
> A esta se le añade otro archivo llamado `.gitignore` que nos deja hacer lo contrario, ignorar ciertos archivos que no queramos llevar a github --Bien porque sea sensible (códigos, contraseñas, etc.) o porque pueda pesar mucho--

[FUENTE](https://www.youtube.com/shorts/EqyY4nMzo_w)

**Comment by mmasias** [comment]
Porfa, me lo recuerdas y lo comentamos en clase!
